R/execute.R

```diff
@@ -1,69 +1,109 @@
+#' Execute a MODFLOW model
 #' 
-#' \code{run_modflow} runs a MODFLOW model.
+#' These functions execute MODFLOW models.
 #' 
-#' @param file path to name file; typically '*.nam'
-#' @param executable name of the MODFLOW executable to use; if not provided, the executable distributed with RMODFLOW is used, corresponding to version, machine and sysname
-#' @param version MODFLOW version to use; 2005 (default) is currently the only option
-#' @param par vector of parameter value file parameter values to run the model with
-#' @param verbose logical; should the terminal output be printed to the R console? Defaults to TRUE
-#' @param convergence logical; should convergence be checked? If TRUE (default), a logical is returned
-#' @param cvg_message character denoting the message in the terminal output used to check for convergence
-#' @return if convergence = TRUE, a logical depending on whether the model converged. Otherwise NULL.
-#' @rdname rmf_run_modflow
-#' @method rmf_run_modflow character
+#' @param code Name of the MODFLOW variant to use, or path to the executable.
+#' @param evaluate Vector of PVAL file parameter values to evaluate. This should
+#'   be a named vector if not all parameters are provided in their order of
+#'   occurrence, where the names (can be regular expressions to) match the
+#'   parameter names. Parameters that are not mentioned take the value from the
+#'   PVAL file. If `NULL` (default), no values are changed in the original
+#'   PVAL file.
+#' @param convergence Character. The message in the terminal output used
+#'   to check for convergence.
+#' @return Invisible list with start and end time, elapsed run time, a logical
+#'   indicating normal termination, and the stdout output, when done for an on
+#'   disk model. Full `modflow` object including all results otherwise (in
+#'   memory model).
 #' @export
-rmf_run_modflow.character <- function(file,
-                                      executable = NULL,
-                                      version = 2005,
-                                      par = NULL,
-                                      verbose = TRUE,
-                                      convergence = TRUE,
-                                      cvg_message = 'Normal termination') {
-  
-  # select appropriate executable
-    if(is.null(executable)) {
-      if(version == '2005' | version == 2005) {
-        executable <- system.file(paste0('bin/MODFLOW-2005_v.1.11.00_',Sys.info()['sysname'],'_',Sys.info()['machine'],'.exe'),package='RMODFLOW')
-      }
-    }
+#' @seealso
+#' [rmf_install()] for external code installation,\cr
+#' [rmf_analyze()] for local sensitivity analysis, and\cr
+#' [rmf_optimize()] for local calibration.
+rmf_execute <- function(...) {
+  UseMethod('rmf_execute')
+}
+
+#' @rdname rmf_execute
+#' @param path Path to the NAM file. Typically with extension `.nam`.
+#' @export
+rmf_execute.character <- function(
+  path,
+  code = "2005",
+  evaluate = NULL,
+  backup = FALSE,
```

Comment:
Will add the documentation. Backup is the same as for `rmf_analyze()` (create backup of pval file that is already on disk, or not), but has different default value here. The `ui` parameter is meant to override the `RMODFLOW.ui` option. Setting it to `"none"` suppresses MODFLOW output, which is done internally in `rmf_analyze()` and `rmf_optimize()` now. The `RMODFLOW.ui` option was also there to set the verbosity of the package in general, but since I moved the UI functions to {rui}, I have to think a bit about how to do that (this should probably go to a {rui} feature request).